tensorflow/core/kernels/lrn_op.cc

```diff
@@ -358,13 +360,15 @@ struct LaunchLRNGrad<CPUDevice, T> {
           }
           norm = alpha_ * norm + bias_;
           DCHECK_GT(norm, T(1e-6));
+          T pre_computed_pow = Eigen::numext::pow(norm, -beta_);
+          T activations_div_norm = alpha_beta_2_ * activations(i, j) / norm;
+          T gs = grads_shaped(i, j);
           for (int64 k = depth_begin; k < depth_end; ++k) {
-            T dyi = T(-2) * alpha_ * beta_ * in_shaped(i, k) *
-                    activations(i, j) / norm;
+            T dyi = in_shaped(i, k) * activations_div_norm;
```

Comment:
My only concern is that moving the division before the in_shaped * activations could result in a loss of precision. Is hoisting the division out of the loop crucial to the speedups you see?

Reply:
I have run CIFAR-10 training on TensorFlow with/without this patch, for a large number of steps, and the accuracy on cifar10_eval afterward was pretty much equal. BTW, I have a follow-up patch on this being finalized, with a vectorized preamble for LRNGrad (this such hot path on training) that also keeps accuracy and brings good speed gains, to be proposed soon.